### PR TITLE
BRC - DRY, force colorization ASAP

### PR DESCRIPTION
**Pull request by Arashiryuu**
- Line 211 is to force them to colorize ASAP, some themes being dumb is why this is being done.
- Condensed a couple of ternary checks into one check.
- MemberList now properly calls its forceUpdate to reflect the plugin's changes.

### CODE REVIEW

File: Plugins/BetterRoleColors/BetterRoleColors.plugin.js [hunk 1]
```diff
@@ -207,6 +207,8 @@ var BetterRoleColors = (() => {
 
                 returnValue.props.onMouseEnter = () => { thisObject.setState({isHovered: true}); };
                 returnValue.props.onMouseLeave = () => { thisObject.setState({isHovered: false}); };
+				
+				if (!Object.prototype.hasOwnProperty.call(thisObject, "isHovered")) thisObject.setState({isHovered: false});
```
Comment (rauenzi) [inline Plugins/BetterRoleColors/BetterRoleColors.plugin.js]: Why is this line needed? Also wouldn't `Object.prototype.hasOwnProperty.call(thisObject, "isHovered")` need to be `Object.prototype.hasOwnProperty.call(thisObject.state, "isHovered")`? (`thisObject` vs `thisObject.state`) Also curious why you chose to use `Object.prototype.hasOwnProperty` rather than on `thisObject.state` directly.

File: Plugins/BetterRoleColors/BetterRoleColors.plugin.js [hunk 2]
```diff
@@ -349,18 +352,17 @@ var BetterRoleColors = (() => {
...
                     };
-                    if (section.props.children) section.props.children.ref = myRef, section.props.children.type = ColoredRoleHeader;
-                    else section.ref = myRef, section.type = ColoredRoleHeader;
+                    children.ref = myRef, children.type = ColoredRoleHeader;
                     return section;
                 });
                 memberList.renderSection.__patched = true;
+				memberList.forceUpdate();
             });
-            MemberList.forceUpdateAll();
```
Comment (rauenzi) [inline Plugins/BetterRoleColors/BetterRoleColors.plugin.js]: Do you know why this doesn't work and has a delay?

File: Plugins/BetterRoleColors/BetterRoleColors.plugin.js (hunk 1 above)
Comment (Arashiryuu) [reply Plugins/BetterRoleColors/BetterRoleColors.plugin.js]: 1 - Forces mentions to re-render with the new styles you set and I don't want to force it if the property exists and is true, whereas without it users including myself would have to mouse over the mention to force it to colorize.
2a - Yeah that was an oversight when I was editing the fork compared to what I had locally.
2b - Preference, I suppose.

File: Plugins/BetterRoleColors/BetterRoleColors.plugin.js (hunk 2 above)
Comment (Arashiryuu) [reply Plugins/BetterRoleColors/BetterRoleColors.plugin.js]: I can't fathom a single reason for it not to, but calling the update like in the PR allows it to work. ¯\_(ツ)_/¯

File: Plugins/BetterRoleColors/BetterRoleColors.plugin.js (hunk 1 above)
Comment (rauenzi) [reply Plugins/BetterRoleColors/BetterRoleColors.plugin.js]: So this is to check if `state` doesn't have an `isHovered` property at all right? When does that occur?

Comment (Arashiryuu) [reply Plugins/BetterRoleColors/BetterRoleColors.plugin.js]: Initial render since it's a property you add to them. They aren't there by default. The `thisObject` seems to refer to the popup from when you click the mention. As such its state normally consists of the `uniqueId` and `isOpen` properties from what I've observed.
